plugins/@grouparoo/calculated-property/src/lib/profileProperty.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  plugin,
+  Profile,
+  ProfileProperty,
+  ProfilePropertyPluginMethod,
+} from "@grouparoo/core";
+import { NodeVM } from "vm2";
+export interface RequiredPropertiesObject {
+  [propertyId: string]: ProfileProperty;
+}
+
+async function calculateProfilePropertyValue(
+  customFunction,
+  profile: Profile
+): Promise<string> {
+  let populatedFunction;
+  try {
+    populatedFunction = await plugin.replaceTemplateProfileVariables(
+      customFunction,
+      profile,
+      false
+    );
+    //returns a string of the entire function
```

Comment:
At this point, the function will be a string with all the variables replaced by their value.  "False" here allows for standard rendering so we can return empty strings for undefined variables.

Reply:
We also want to make sure that users aren't attempting to require any libraries or access env variables... functions doing this would fail when executed, but we want them to fail before even getting to that point.